MovingBlocks/Terasology
# NUI editor: additional settings and finishing touches

### Contains

Various improvements to the NUI screen/skin editors; see rzats/gsoc-nui-editor#10, rzats/gsoc-nui-editor#13 and rzats/gsoc-nui-editor#14.

- Added a 'Preview screen locale' setting - this will change the language of the editor's preview widget (but not the NUI editor or the game itself). Useful for reviewing things such as long UI strings that may break interface rendering!
- Added a "Close" button to the editors (as an alternative to ESC & F10/F11) and a "Delete" option to tree view node context menues (as an alternative to DELETE).

### How to test

Press F10/F11 in the main menu to run the NUI screen/skin editors, then verify that all of the above work as intended :smile:

### Outstanding before merging

- [ ] Implement the features from rzats/gsoc-nui-editor#13 (tree view to widget binding).
- [x] Review the source code's Javadoc, modify and fill stubs where appropriate.
  - [x] Also fix Checkstyle warnings where appropriate.
- [x] Autosave the asset changes to a temporary JSON file, then add an option to apply the changes to the actual asset file.
- [x] Add a popup to warn the user that changes will be lost on asset change.
- [x] Automatically edit newly added attributes.
  - [x] Automatically edit the IDs of newly added widgets.
- [x] A "Debug Tools" menu with options to show different dev tools (console, UI editor, skin editor, BT editor).
- [x] When the editor's strings are somewhat finalized, internationalize them all!
